src/routes/transactions/entities/human-description.entity.ts

```diff
@@ -0,0 +1,25 @@
+import { Token } from '@/domain/tokens/entities/token.entity';
+import {
+  AddressFragment,
+  NumberFragment,
+  ValueType,
+  WordFragment,
+} from '@/domain/human-description/entities/human-description.entity';
+
+export interface RichTokenValueFragment {
+  type: ValueType.TokenValue;
+  value: {
+    amount: string;
+    token: Token | null;
```

Comment:
I would not expose the domain entity directly here to the clients. The main reason is if we refactor this domain entity, we would break the clients.

By having what the clients expect separate from the domain entity we know that by changing the domain entity we also need to change the mapping.

In order to reduce nesting I would consider for now only what the clients for rendering. Suggestion:

```typescript
export interface RichTokenValueFragment {
  type: ValueType.TokenValue;
  amount: string;
  symbol: string;
  logoUri: string;
}
```

Reply:
With domain entity do you mean the `ValueType` enum or that this type has the same "structure" as the fragment from the domain entity? I went with a flat structure now and only return what is needed as you suggested.